packages/@aws-cdk/aws-appmesh/lib/gateway-route.ts

```diff
@@ -0,0 +1,390 @@
+   */
+  public abstract bind(scope: cdk.Construct): GatewayRouteSpecConfig;
+}
+
+class HttpGatewayRouteSpec extends GatewayRouteSpec {
+  /**
+   * The criterion for determining a request match for this GatewayRoute.
+   *
+   * @default - matches on '/'
+   */
+  readonly match?: HttpGatewayRouteMatch;
+  /**
+   * The VirtualService this GatewayRoute directs traffic to
+   */
+  readonly routeTarget: IVirtualService;
+
+  /**
+   * Type of route you are creating
+   */
+  readonly routeType: Protocol;
+
+  constructor(props: HttpRouteSpecProps, protocol: Protocol.HTTP | Protocol.HTTP2) {
+    super();
+    this.routeTarget = props.routeTarget;
+    this.routeType = protocol;
+    this.match = props.match;
+  }
+  public bind(_scope: cdk.Construct): GatewayRouteSpecConfig {
+    const prefixPath = this.match ? this.match.prefixPath : '/';
+    if (prefixPath[0] != '/') {
+      throw new Error('Prefix Path must start with \'/\'');
+    }
+    const httpConfig: CfnGatewayRoute.HttpGatewayRouteProperty = {
+      match: {
+        prefix: prefixPath,
+      },
+      action: {
+        target: {
+          virtualService: {
+            virtualServiceName: this.routeTarget.virtualServiceName,
+          },
+        },
+      },
+    };
+    return {
+      httpSpecConfig: this.routeType === Protocol.HTTP ? httpConfig : undefined,
+      http2SpecConfig: this.routeType === Protocol.HTTP2 ? httpConfig : undefined,
+    };
+  }
+}
+
+class GrpcGatewayRouteSpec extends GatewayRouteSpec {
+  /**
+   * The criterion for determining a request match for this GatewayRoute.
+   *
+   * @default - no default
+   */
+  readonly match: GrpcGatewayRouteMatch;
+  /**
+   * The VirtualService this GatewayRoute directs traffic to
+   */
+  readonly routeTarget: IVirtualService;
+
+  constructor(props: GrpcRouteSpecProps) {
+    super();
+    this.match = props.match;
+    this.routeTarget = props.routeTarget;
+  }
+
+  public bind(_scope: cdk.Construct): GatewayRouteSpecConfig {
+    return {
+      grpcSpecConfig: {
+        action: {
+          target: {
+            virtualService: {
+              virtualServiceName: this.routeTarget.virtualServiceName,
+            },
+          },
+        },
+        match: {
+          serviceName: this.match.serviceName,
+        },
+      },
+    };
+  }
+}
+
+/**
+ * GatewayRoute represents a new or existing gateway route attached to a VirtualGateway and Mesh
+ *
+ * @see https://docs.aws.amazon.com/app-mesh/latest/userguide/gateway-routes.html
+ */
+export class GatewayRoute extends cdk.Resource implements IGatewayRoute {
+  /**
+   * Import an existing GatewayRoute given an ARN
+   */
+  public static fromGatewayRouteArn(scope: Construct, id: string, gatewayRouteArn: string): IGatewayRoute {
+    return new ImportedGatewayRoute(scope, id, { gatewayRouteArn });
+  }
+
+  /**
+   * Import an existing GatewayRoute given its name
+   */
+  public static fromGatewayRouteName(
```

Comment:
If you have some many parameters here, we usually instead introduce a `fromGatewayRouteAttributes(Constrict, string, GatewayAttributes)` method, where `GatewayAttributes` is:

```ts
export interface GatewayAttributes {
  readonly mesh: IMesh;
  readonly virtualGateway: IVirtualGateway;
  readonly gatewayRouteName: string;
}
```

Reply:
I've changed this and other resources to have `fromAttributes`